packages/js/wasm/src/imports.ts

```diff
@@ -215,9 +246,21 @@ export const createImports = (config: {
         const msg = readString(memory.buffer, msgPtr, msgLen);
         const file = readString(memory.buffer, filePtr, fileLen);
 
-        abort(
-          `__wrap_abort: ${msg}\nFile: ${file}\nLocation: [${line},${column}]`
-        );
+        let code = WrapErrorCode.WASM_INVOKE_ABORTED;
+
+        // check if error was thrown in bindings
+        if (
+          file.startsWith("~lib/@polywrap/wasm-as/msgpack/") ||
+          file.startsWith("src/wrap/")
+        ) {
+          code = WrapErrorCode.WASM_SERIALIZATION_ERROR;
```

Comment:
Hmmmm, is there maybe a more consistent way of detecting this? This feels wrong...

Reply:
It's definitely hacky. It would be nice to hear other ideas.

The other main ways I thought to handle this were:
1. Add a prefix to errors thrown in msgpack or bindings, such as "$wrapCode_42 [message]", that could be stripped and parsed. We could also let wrapper devs use this to set error codes when they throw.
2. I thought to create a special `wrap_abort` method to be called, but I couldn't find a way to get source info manually.
3. We could instead not use a serialization error code at all. In this case, I think it would be a good idea to add a little more information to the errors we throw in msgpack and bindings so that it's more clear to the user that it's a serialization error.

Reply:
I think a better approach to this would be:
- Have the wasm bindings be "WrapError aware".
- Throw the `WASM_SERIALIZATION_ERROR` code from within the wasm bindings.
- Maybe parse & extract the code within the `__wrap_abort` function from the `msg` argument? Not sure this is needed though.

Reply:
Actually, I retract my last comment. I think the `WrapError` should be a client-side only thing (for now). Based on this, the errors should only consist of things that the client can "know" have happened. Based on this, I think the `WASM_SERIALIZATION_ERROR` is inherently "flawed" since it steps over the wasm boundary, requiring the client to know what happened within the wasm module's runtime.

When you look at the full set, you can see that all of them (except the last) are client-side things, but the last one requires wasm-side context:
```
WASM_INVOKE_ABORTED
WASM_SUBINVOKE_ABORTED
WASM_INVOKE_FAIL
WASM_MODULE_NOT_FOUND
WASM_STATE_ERROR
WASM_SERIALIZATION_ERROR
```

I think we can just remove the `WASM_SERIALIZATION_ERROR` entirely, because the error message will have enough information for a user to debug further (using the serialization context stack we have already implemented).

Thoughts?

Reply:
This sounds fantastic. Apologies for taking so long to respond!

Reply:
Hey no problem, also I didn't see your comment above originally when I typed my 2nd response (didn't reload my page). I really like your idea about "Add a prefix to errors thrown in msgpack or bindings, such as "$wrapCode_42 [message]", that could be stripped and parsed. We could also let wrapper devs use this to set error codes when they throw."

I think this would be a great thing to add in the future so we can easily have new error codes "cross the boundary" and make it more part of the WRAP interoperability standard (similar to HTTP error codes). For now just starting with this as a client-side only thing is an easy first step.